tikv/region.go

```diff
@@ -176,10 +177,147 @@ type RegionOptions struct {
 	RegionSize int64
 }
 
-type RegionManager struct {
-	storeMeta  metapb.Store
-	mu         sync.RWMutex
-	regions    map[uint64]*regionCtx
+type RegionManager interface {
+	GetRegionFromCtx(ctx *kvrpcpb.Context) (*regionCtx, *errorpb.Error)
+	CheckLeader(ctx *kvrpcpb.Context) *errorpb.Error
```

Comment:
We can do `CheckLeader` inside `GetRegionFromCtx` and remove `CheckLeader` method from `RegionManager`.